subsys/bluetooth/mesh/dfu_srv.c

```diff
@@ -87,15 +89,18 @@ static void apply_rsp_sent(int err, void *cb_params)
 	if (!srv->cb->apply || srv->update.idx == UPDATE_IDX_NONE) {
 		srv->update.phase = BT_MESH_DFU_PHASE_IDLE;
 		store_state(srv);
+		LOG_DBG("Prerequisites for apply callback are wrong");
 		return;
 	}
 
+	store_state(srv);
+
 	err = srv->cb->apply(srv, &srv->imgs[srv->update.idx]);
 	if (err) {
 		srv->update.phase = BT_MESH_DFU_PHASE_IDLE;
+		store_state(srv);
+		LOG_DBG("Application apply callback failed");
```

Comment:
Perhaps use LOG_WRN or LOG_ERR here? 

Also, perhaps add the error code in the last log message? (I see that this isn't generally done in the rest of the file, but could be an improvement!)

Reply:
I do not think this is critical for stack to generate warning or even error here. Application or test aka application can manage its own errors by self. As a debug information, this should be sufficient.

Reply:
Ok!